sessionctx/variable/session.go

```diff
@@ -344,3 +344,18 @@ func (sc *StatementContext) AppendWarning(warn error) {
 	}
 	sc.mu.Unlock()
 }
+
+// HandleTruncate ignores or returns the error based on the StatementContext state.
+func (sc *StatementContext) HandleTruncate(err error) error {
+	if err == nil {
```

Comment:
There may be many kinds of truncate error, check it here need to add many dependencies creates dependency cycle, so 
we just let caller make sure it's a truncate error.